lib/backend/block_affinity.go

```diff
@@ -44,7 +44,19 @@ func (key BlockAffinityKey) asEtcdKey() (string, error) {
 }
 
 func (key BlockAffinityKey) asEtcdDeleteKey() (string, error) {
-	return key.asEtcdKey()
+	// Must specify at least a host to remove.
+	if key.Host == "" {
+		return "", common.ErrorInsufficientIdentifiers{}
+	}
+	k := "/calico/ipam/v2/host/" + key.Host
```

Comment:
Yes I was thinking the same for WorkloadEndpoints where I think Tom needs a way to delete a Workload and all of it's endpoints.

I was thinking that adding a Workloads interface where we probably just have a subset of available commands.